src/components/NavigationModal.js

```diff
@@ -0,0 +1,83 @@
+const React = require('react');
+const _merge = require('lodash/merge');
+
+const Icon = require('./Icon');
+
+const StyleConstants = require('../constants/Style');
+
+const NavigationModal = React.createClass({
+  propTypes: {
+    componentStyle: React.PropTypes.object,
+    iconSize: React.PropTypes.number,
+    iconStyle: React.PropTypes.object,
+    iconType: React.PropTypes.string,
```

Comment:
Also these are not required and have no default values and your example in demo doesn't supply them.  I'm also not seeing how you protect against using them below in the render function for the Icon component.